param/__init__.py

```diff
@@ -1174,11 +1178,27 @@ class ObjectSelector(SelectorBase):
 
     __slots__ = ['objects', 'compute_default_fn', 'check_on_set', 'names']
 
-    # ObjectSelector is usually used to allow selection from a list of
+    # Selector is usually used to allow selection from a list of
     # existing objects, therefore instantiate is False by default.
-    def __init__(self, default=None, objects=None,  instantiate=False,
+    def __init__(self, objects=None, default=None, instantiate=False,
```

Comment:
This order is indeed better but in theory could break anyone passing in the arguments by position. This causing an issue is very unlikely though as I don't think I've ever seen that style of signature so overall this is better.